kiwi/markup/__init__.py

```diff
@@ -44,8 +44,11 @@ def new(description: str, name: str='any'):  # noqa: E252
             markup = Markup._load_markup_by_name(name, description)
             log.info('Support for multiple markup descriptions available')
         except KiwiAnyMarkupPluginError:
+            log.warn('Support for non-XML formatted config files requires '
+                     'the Python anymarkup module.')
+
             markup = Markup._load_markup_by_name('xml', description)
-            log.info('Support for XML markup available')
+            log.info('Processing {} as XML'.format(repr(description)))
```

Comment:
Inspired by the python class we can also write this as

```python
f'Processing {description!r} as XML'
```

:smile:

Reply:
Well actually we should probably be using some form of lazy formatting, or at least that is what pylint complains about when I used f-strings with log messages, so that we don't render the string as a log.info() argument up front but rather use something like log.info(fmt_str, arg1, arg2, ...).

My understanding is the lazy formatting pattern is recommended as it both allows for easy integration with localisation tools, as the format string can be replaced by the localised version taking the same args without significant code impact, and defers rendering the formatted string until when it is actually being logged, which may never actually occur if filter/config settings mean that the log message is skipped...

Whereas f-strings are rendered as is where they occur, which doesn't work well for localisation. ;-)

Reply:
Oh yes, this is passed to the logging facility. You are right the linter complains with fstrings. Sorry my bad